dali/pipeline/workspace/iteration_data.h

```diff
@@ -74,8 +76,11 @@ class OperatorTraces {
 struct IterationData {
   int64_t iteration_index = 0;
   OperatorTraces operator_traces;
+  std::shared_ptr<Checkpoint> checkpoint;
```

Comment:
I used `shared_ptr` here because it allows `Checkpoint` to be only forward-declared. Using `unique_ptr` wouldn't be much more elegant and would require IterationData to be DLL_PUBLIC (meh) and a separate translation unit for the destructor (blah) or a unique_ptr with a custom deleter (blah).